### PR TITLE
fix: save int16 audio with soundfile to avoid torchaudio clipping

### PR DESCRIPTION
**Pull request by laoxuaichirou**
# Fix: corrupted int16 audio when saving with torchaudio 2.11+ / torchcodec

## Problem

`torchaudio.save()` on recent torchaudio builds (>= 2.9, notably 2.11 with the
torchcodec backend) writes **int16** tensors incorrectly: every intermediate
sample value is clamped/pushed to full-scale (±1.0), producing loud static /
noise instead of the expected waveform.

Reproduction (torchaudio 2.11, torchcodec 2.11 backend):

```python
import torch, torchaudio, soundfile as sf
wav = torch.tensor([32767, 0, 32767, 16000, -8000, 22937])
torchaudio.save("t_a.wav", wav, 16000)
sf.write("t_b.wav", wav.numpy(), 16000, subtype="PCM_16")
# read back: t_a.wav -> [1, 0, 1, 0.9999, -1, 0.9999]  (all clipped to full scale)
# read back: t_b.wav -> [1, 0, 1, 0.488, -0.244, 0.7]  (correct)
```

Symptoms in IndexTTS 2.5 output: waveform RMS ≈ 0.95, >90% of samples at
full-scale amplitude, no pauses/silence, audio plays as noise/clipping.
Same seed + same weights, only the save call differs.

## Fix

Replace the two `torchaudio.save()` calls in `indextts/infer_v2_5.py` with
`soundfile` (`sf.write(..., subtype="PCM_16")`), which handles int16 correctly
and is already a transitive dependency of the project stack.

- low-vram branch (~line 542): the tensor is float32 in [-1, 1], write as-is.
- main branch (~line 911): the tensor is int16, move to CPU and transpose before
  writing (soundfile expects channels-last).

Add `soundfile>=0.12.1` to `pyproject.toml`.

## Verification

- After the fix: waveform RMS ≈ 0.10, 0% full-scale samples, ~30% pause ratio,
  matching the reference prompt audio magnitude.
- All existing pytest suites still pass (`pytest -m "not gpu"`).

## Notes

The torchaudio 2.8.* pin in pyproject.toml is unaffected; this change also makes
synthesis robust on newer torchaudio builds where the old pin is unavailable
(e.g. Windows wheels for torchaudio 2.11).

### CODE REVIEW

**Comment by nanaoto** [comment]
fixed in #773
